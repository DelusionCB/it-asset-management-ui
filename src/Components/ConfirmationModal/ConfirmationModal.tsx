import React from 'react';
import {Button, Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
import './index.scss'
import {ConfirmationModalPropTypes} from '../../Types/types.components';

function ConfirmationModal ({onConfirm, onCancel, values, t, type}: ConfirmationModalPropTypes): JSX.Element {
    return (
        <Modal centered={true} className='confirmation-modal' isOpen={true} toggle={onCancel}>
            <ModalHeader className='-header' tag='h1'>{t('modal.header')}</ModalHeader>
            <ModalBody className='-body'>
                <div>
                    <div>
                        {t(`modal.body.${type}`, {value: values.name})}
                    </div>
                </div>
            </ModalBody>
            <ModalFooter className='-footer'>
                <Button className='-do' onClick={onConfirm}>
                    {t('modal.doAction')}
                </Button>
                <Button className='-cancel' onClick={onCancel}>
                    {t('modal.cancelAction')}
                </Button>
            </ModalFooter>
        </Modal>
    );
}

export default ConfirmationModal;
