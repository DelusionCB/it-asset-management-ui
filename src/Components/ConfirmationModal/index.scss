
.confirmation-modal {

    .-header {
        background-color: var(--primary-color);
        padding: 10px;
        border-bottom: 2px solid var(--secondary-color);
        color: white;
    }

    .-body {
        font-size: 1.5rem;
    }

    .-footer {
        button {
            font-size: 1.25rem;
            border: 2px solid black;
            &:hover, &:focus, &:active {
                border: 2px solid var(--secondary-color);
            }
        }
        .-do {
            background-color: var(--danger-color);
        }
        .-cancel {
            background-color: var(--primary-color);
        }
    }
}
