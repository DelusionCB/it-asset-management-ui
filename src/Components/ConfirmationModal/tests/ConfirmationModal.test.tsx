import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ConfirmationModal from '../ConfirmationModal';
import '@testing-library/jest-dom';

// Mock for react-dom as Modal -components require it
jest.mock('react-dom', () => {
    const original = jest.requireActual('react-dom');

    return {
        ...original,
        createPortal: (node) => node,
    };
});

describe('<ConfirmationModal />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <ConfirmationModal t={jest.fn()} onConfirm={jest.fn()} onCancel={jest.fn()} type='application' values={{}} />,
        );
    });

    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
