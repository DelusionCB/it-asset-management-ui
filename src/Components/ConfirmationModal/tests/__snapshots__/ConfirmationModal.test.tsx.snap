// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> Snapshot renders correctly 1`] = `
<div
  className=""
>
  <div
    className="modal fade"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    role="dialog"
    style={
      {
        "display": "block",
      }
    }
    tabIndex="-1"
  >
    <div
      className="modal-dialog confirmation-modal modal-dialog-centered"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="-header modal-header"
        >
          <h1
            className="modal-title"
          />
        </div>
        <div
          className="-body modal-body"
        >
          <div>
            <div />
          </div>
        </div>
        <div
          className="-footer modal-footer"
        >
          <button
            className="-do btn btn-secondary"
            onClick={[Function]}
            type="button"
          />
          <button
            className="-cancel btn btn-secondary"
            onClick={[Function]}
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-backdrop fade"
  />
</div>
`;
